docs/content/docs/data-structures/tree.md

```diff
@@ -485,61 +498,98 @@ For the meaning of "simultaneously", see [Types of distributed data structures](
 
 ### Events
 
-The `SharedTree` object supports two events: `beforeChange` and `afterChange`. Your code can create handlers for these events using the utility class `Tree`. See [Tree utility APIs](#tree-utility-apis).
+`SharedTree` supports two node level events: `nodeChanged` and `treeChanged`. Your code can create handlers for these events using the utility class `Tree`. See [Tree utility APIs](#tree-utility-apis).
+
+Additionally, the `TreeView` object includes 3 events that operate over the whole tree. These are `rootChanged`, `afterBatch`, and `commitApplied`.
+
+`rootChanged` fires when the root field (the field that contains the root node) changes. That is, if a new root node is assigned or the schema changes. This will not fire when the node itself changes.
+
+`afterBatch` fires when a batch of changes has been processed by the Fluid Framework and the `TreeView` object is updated.
+
+`commitApplied` fires whenever a local change is applied outside of a transaction or when a local transaction is committed. This is used to get `Revertible` objects to put on the undo stack. See [Undo/Redo support](#undoredo-support) and [Transactions](#transactions).
+
+### Undo/Redo support
+
+`SharedTree` makes creating an undo and redo stack very simple. By listening for the `commitApplied` event on the `TreeView` object, you can get a `Revertible` object.
+`Revertible` objects come in three flavors:
```

Comment:
Talking about flavors of Revertibles is at odds with our API since it is the commits that are marked with those flavors.

Reply:
Fixed